pandas/core/groupby/groupby.py

```diff
@@ -3212,6 +3212,9 @@ def ngroup(self, ascending: bool = True):
         would be seen when iterating over the groupby object, not the
         order they are first observed.
 
+        If a group would be excluded (due to null keys) then that
+        group is labeled as np.nan. See examples below.
```

Comment:
I think it could be simpler to understand if we just simply say something like `Groups where the key is missing are skipped from the count, and their value will be 'NaN'`.

Reply:
I'd prefer NA as opposed to "missing". There are various reasons a value can be NA, missing is but one of them. Also recommend using NA over NaN or null as it matches `dropna` and `pd.NA`.

Reply:
Fine by me. But in this case if we say `NA` it may sound like `pandas.NA`, and if the value is `None` or`NaN` it will also be skipped. So, missing seemed more appropriate to me.

Reply:
Maybe `not available` instead of `NA` is more generic than missing, and doesn't sound like we are talking about `pandas.NA`?

Reply:
Sorry - that was poorly written. My main reason is to connect it with the `dropna` argument. I threw in `pd.NA` as an indicator that we use NA with somewhat good consistency. E.g. dropna, pd.NA, isna, use_inf_as_na, fillna, na_filter. I think we should maintain this.

Reply:
I adjusted to "Groups with missing keys (where `pd.isna()` is True) will be labeled with `NaN` and will be skipped from the count." Definitely this is the right direction, I'm still just not sure about the "missing" language

I agree with

> But in this case if we say NA it may sound like pandas.NA, and if the value is None orNaN it will also be skipped.

so I avoided saying simply "groups with NA keys...". I think my phrasing is maybe a little longer, but extremely precise and fairly easy to understand. Another option I would also be satisfied with is "Groups with NA keys (where `pd.isna()` is True)"

Reply:
Thanks - this looks great to me.